proposed/2022/package-scoring-v1.md

```diff
@@ -0,0 +1,222 @@
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+Package scoring will contain four unique categories to start off with. The categories are the following:
+
+- Popularity - How popular a package is & recognized in the ecosystem.
+- Quality - The completeness of a package following best practices & providing documentation.
+- Maintenance - The state of maintenance for a package based on it's update cadence.
+- Security - The sense of trust based on a package being free of known security vulnerabilities, license risk, and current supply chain risk.
+
+Each category has the potential to total up to 100 points.
+
+- Categories in which the package is **in control** can always reach the maximum amount of points.
+- Categories in which the package is **not in control** will be determined by the state of the ecosystem.
+
+EX: A package is in control of providing ample metadata to ensure the quality of the package reaches the full score. A package is not in control of how popular it may be however.
+
+To scale package scoring for the future, each score will be comprised of an analysis of package performance and potential package/dependency issues. Each of these issues will be categorized into one of these four categories and empower the author and consumer to resolve these issues to the best of their ability and control.
+
+#### Issues
+
+As a means to scale package scoring to it's potential through continuous iteration, each category will have its own set of issues that a user can be aware of when browsing through the package. Keeping the same idea of **being in control** or **not in control** also applies to package issues.
+
+For issues which can be addressed, an issue will include an empowering message regarding how one can take action to help address the issue.
+
+EX: A package might be missing a README which a developer may not be able to easily get started with the package. They might be able to then contribute a new issue or even the inclusion of the README file to the NuGet package if it's open source.
+
+**Issue Examples:**
+
+- Invalid .nuspec / .csproj
```

Comment:
.csproj is not known by NuGet.org and is lost at the time of "pack". I don't see how this could be a dimension considered by NuGet.org or by a CLI command that operates offline, against the .nupkg file.

Reply:
I'd probably remove this altogether. You can't publish a package with an invalid nuspec.